src/main/java/racingcar/domain/Car.java

```diff
@@ -0,0 +1,44 @@
+package racingcar.domain;
+
+import racingcar.dto.RoundResponse;
+
+import java.util.Objects;
+
+public class Car {
+    private static final Integer START_SCORE = 0;
+
+    private final Name name;
+    private Integer score;
+
+    private Car(final String name) {
+        this.name = Name.create(name);
+        this.score = START_SCORE;
+    }
+
+    public static Car create(final String name) {
+        return new Car(name);
+    }
+
+    public void move() {
+        MovementCondition condition = MovementCondition.create();
+        if (condition.canMovable()) {
+            score++;
+        }
+    }
+
+    public boolean isSameScore(final Integer score) {
+        return Objects.equals(this.score, score);
+    }
+
+    public RoundResponse generateRoundResponse() {
+        return new RoundResponse(name.getName(), score);
+    }
```

Comment:
car에서 Response를 생성하는 것 보다 Response에서 car를 받아 내부적으로 생성하는 로직으로 사용하는건 어떻게 생각하시나요?

ex
```suggestion
   public static RoundResponse of(Car car) 
      return new RoundResponse(...);
    }
```

Reply:
Car 객체의 getter 참조(값 타입의 이동)를 최대한 줄이기 위해 위와 같이 설계했습니다.
필연적으로 이름은 Dto에 넘어가야 했던 부분이고, score같은 경우 도메인에서 선언했을 때,
score 자체를 Dto에 넘기지 않아도 되는 설계가 되었습니다.

일급 컬렉션을 적용하면서 객체간 의존이 생기는데, 이 의존을 getter없이 풀어내면서 고민했던 로직이었습니다..!
다만 말씀주신 로직이 더욱 구조상 깔끔한 것은 누가 봐도 사실인 것 같습니다.
getter 지양 설계에 너무 집착한 나머지, 좋은 설계를 조금 놓친 것 같아요.

좋은 리뷰 감사합니다 :)